pycqed/measurement/sweep_functions.py

```diff
@@ -303,3 +303,107 @@ def __init__(self, sweep_function, values, name=None, parameter_name=None,
 
     def default_param_name(self):
         return f'index of {self.sweep_function.parameter_name}'
+
+
+class MajorMinorSweep(Soft_Sweep):
+    """
+    A soft sweep function that combines two sweep function such that the
+    major sweep function takes only discrete values from a given set while
+    the minor sweep function takes care of the difference between the
+    discrete values and the desired sweep values.
+
+    (further parameters as in multi_sweep_function)
+    :param major_sweep_function: (obj) a soft sweep function or QCoDeS
+        parameter to perform large steps of the sweep parameter
+    :param minor_sweep_function: (obj) a soft sweep function or QCoDeS
+        parameter to perform small steps of the sweep parameter
+    :param major_values: (array, list) allowed values of the
+        major_sweep_function
+    """
+
+    def __init__(self,
+                 major_sweep_function,
+                 minor_sweep_function,
+                 major_values,
+                 name=None,
+                 parameter_name=None,
+                 unit=None):
+        super().__init__()
+
+        self.major_sweep_function = \
+            mc_parameter_wrapper.wrap_par_to_swf(major_sweep_function) \
+                if isinstance(major_sweep_function, qcodes.Parameter) \
+                else major_sweep_function
+        self.minor_sweep_function = \
+            mc_parameter_wrapper.wrap_par_to_swf(minor_sweep_function) \
+                if isinstance(minor_sweep_function, qcodes.Parameter) \
+                else minor_sweep_function
+        if self.major_sweep_function.sweep_control != 'soft' or \
+                self.minor_sweep_function.sweep_control != 'soft':
+            raise ValueError('Offset_Sweep: Only software sweeps supported')
+        self.sweep_control = 'soft'
+        self.major_values = np.array(major_values)
+        self.parameter_name = self.major_sweep_function.parameter_name \
+            if parameter_name is None else parameter_name
+        self.name = self.major_sweep_function.name if name is None else name
+        self.unit = self.major_sweep_function.unit if unit is None else unit
+
+    def prepare(self, *args, **kwargs):
+        self.major_sweep_function.prepare(*args, **kwargs)
+        self.minor_sweep_function.prepare(*args, **kwargs)
+
+    def finish(self, *args, **kwargs):
+        self.major_sweep_function.finish(*args, **kwargs)
+        self.minor_sweep_function.finish(*args, **kwargs)
+
+    def set_parameter(self, val):
+        # find the closes allowed value of the major_sweep_function
+        ind = np.argmin(np.abs(self.major_values - val))
+        mval = self.major_values[ind]
+        self.major_sweep_function.set_parameter(mval)
+        # use the minor_sweep_function to bridge the difference to the
+        # target value
+        self.minor_sweep_function.set_parameter(val - mval)
+
+
+class FilteredSweep(multi_sweep_function):
```

Comment:
I wanted to give the user the possibility to directly use a filtered sweep in combination with multiple sweep functions without adding a extra hierarchy level (to keep the hierarchy of sweep functions flat, i.e., to avoid thing like MC.sweep_functions[1].sweep_function.sweep_functions[0] etc).

The code in QE currently does not make use of this: for the sake a unified treatment of all kinds of soft sweep functions, I accepted the drawback of adding another hierarchy layer in the code in QE). However, I think it does not hurt to give the user the possibility to use the FilteredSweep directly as a multi sweep function. Or do you think it has significant drawbacks?

Reply:
I think my comment was more about `multi_sweep_function` in general -- the functionality of `multi_sweep_function` seems to be that it sets the same sweeppoint value to multiple sweep functions. For parallel sweeps, functionality to set different values to different sweep functions seems much more useful.

Reply:
Yes, I use multi sweep function a lot in the ParallelLOSweepExperiment code on the S17 branch. Will be one of the next PRs, but I cannot create that PR yet due to conflicts with open PRs.